src/Command/BuyItemCommand.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace VendingMachine\Command;
+
+use VendingMachine\Coordinator\PaymentCoordinator;
+use VendingMachine\Repository\ItemRepository;
+use VendingMachine\Request\BuyItemRequest;
+use VendingMachine\Response\ConsoleResponse;
+
+class BuyItemCommand
+{
+    public function __construct(
+        private ItemRepository $itemRepository,
+        private PaymentCoordinator $paymentCoordinator,
+        private ConsoleResponse $response
+    ) {
+    }
+
+    public function __invoke(BuyItemRequest $request): ConsoleResponse
+    {
+        $item = $request->item();
+        $moneys = $request->moneyCollection();
+
+        $availableItem = $this->itemRepository->getItemBySelector($item);
+
+        if (!$availableItem) {
+            throw new \InvalidArgumentException(sprintf('Item %s is not available', $item));
```

Comment:
I thinking about this exception - exceptions should be for unexpected situations (like disconnect, file not found, index larger than size of array etc) and not availabled item is an expected situation by business flow so I think it should be as:

```php
return new ItemNotAvailableResponse($item);
```

PS. As functional programmer, I don't like exceptions ; -)